libraries/core/src/main/kotlin/io/element/android/libraries/core/extensions/Result.kt

```diff
@@ -7,6 +7,61 @@
 
 package io.element.android.libraries.core.extensions
 
+import kotlin.coroutines.cancellation.CancellationException
+
+/**
+ * Can be used to catch exceptions in a block of code and return a [Result].
+ * If the block throws a [CancellationException], it will be rethrown.
+ * If it throws any other exception, it will be wrapped in a [Result.failure].
+ *
+ * [Error]s are not caught by this function, as they are not meant to be caught in normal application flow.
+ */
+inline fun <T> catchingExceptions(
```

Comment:
To be closer to `runCatching`, maybe rename to `runCatchingExceptions`?

Also I am wondering if we could still have a safe mode (for production?) where we still catch `Error`s. Here we take the risk to have crashes on production. Maybe we could have only this on debug build, so that we have time to fix existing problems?

Reply:
I don't think we can, not easily 🫤 . This is in a java library module so we don't have the concept of `BuildConfig` to check if it's a debug or release version and we can't inject the `BuildMeta` here either.